File: 基礎題庫/Contents/b762/b762.py
import sys

for n in sys.stdin:
    try:
        n = int(n.strip())
    except:
        continue
    
    kill = assist = die = flag = 0
    for i in range(n):
        temp = input().strip()

        if temp == "Get_Kill":
            kill += 1
            flag += 1

            if flag == 1 or flag == 2:
                print("You have slain an enemie.")
            elif flag == 3:
                print("KILLING SPREE!")
            elif flag == 4:
                print("RAMPAGE~")
            elif flag == 5:
                print("UNSTOPPABLE!")
            elif flag == 6:
                print("DOMINATING!")
            elif flag == 7:
                print("GUALIKE!")
            else:
                print("LEGENDARY!")
        elif temp == "Get_Assist":
            assist += 1
        elif temp == "Die":
            if flag < 3:
                print("You have been slained.")
            else:
                print("SHUTDOWN.")
            
            die += 1
            flag = 0
        
    print("%d/%d/%d" % (kill, die, assist))

File: 基礎題庫/Contents/b762/b762.c
#include<stdio.h>
#include<string.h>

int main(void)
{
	int n;
	char record[11];
	
	while(scanf("%d", &n) != EOF)
	{
		int kill = 0;
		int assist = 0;
		int die = 0;
		int flag = 0;
		
		for(int i = 0; i < n; i++)
		{
			scanf("%s", record);
			
			if(!strcmp(record, "Get_Kill"))
			{
				kill++;
				flag++;
				
				switch(flag)
				{
					case 1:
					case 2:
						printf("You have slain an enemie.\n");
						break;
					case 3:
						printf("KILLING SPREE!\n");
						break;
					case 4:
						printf("RAMPAGE~\n");
						break;
					case 5:
						printf("UNSTOPPABLE!\n");
						break;
					case 6:
						printf("DOMINATING!\n");
						break;
					case 7:
						printf("GUALIKE!\n");
						break;
					default:
						printf("LEGENDARY!\n");
						break;
				}
			}
			else if(!strcmp(record, "Get_Assist"))
				assist++;
			else if(!strcmp(record, "Die"))
			{
				if(flag < 3)
					printf("You have been slained.\n");
				else
					printf("SHUTDOWN.\n");
				
				die++;
				flag = 0;
			}
		}
		
		printf("%d/%d/%d\n", kill, die, assist);
	}
	
	return 0;
}
